core/shared/src/main/scala/com/monovore/decline/Argument.scala

```diff
@@ -181,6 +182,23 @@ object Argument extends PlatformArguments {
       } catch { case _: IllegalArgumentException => Validated.invalidNel(s"Invalid UUID: $string") }
     }
 
+  implicit val readDuration: Argument[Duration] =
+    from("duration") { string =>
+      try Validated.Valid(Duration(string))
+      catch { case _: NumberFormatException => Validated.invalidNel(s"Invalid Duration: $string") }
+    }
+
+  implicit val readFiniteDuration: Argument[FiniteDuration] =
+    new Argument[FiniteDuration] {
+      override def read(string: String): ValidatedNel[String, FiniteDuration] =
+        readDuration.read(string) andThen {
+          case fd: FiniteDuration => Validated.Valid(fd)
+          case _ => Validated.invalidNel(s"Invalid Duration: $string is not finite")
+        }
+
+      override def defaultMetavar: String = readDuration.defaultMetavar
```

Comment:
Not sure if you want to have `finite-duration` here instead of just the original `duration`?

Reply:
I think just `duration` is fine here as well.

(I try to think of these things from the perspective of the CLI user who has no idea how it's implemented internally, and it's not clear that they'd know or care about the distinction in this case... as a user I'd probably assume that all durations were finite unless it was called out explicitly in the docs.)